db/warehouse/migrate/20240414183410_add_enrollment_slug_to_services.rb

```diff
@@ -0,0 +1,33 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+class AddEnrollmentSlugToServices < ActiveRecord::Migration[6.1]
+  def up
+    safety_assured do
```

Comment:
So this may be a problem for deployment since the tables are locked while this occurs (same in the CustomServiceTypeSlug migration below). Should we try and find a way to reduce service interruption or just plan to deploy during off-hours?

Reply:
We can probably plan for off-hours.  I think the other solution would be to duplicate the table, add the column and then swap in the new table, but while it might not lock the table, it risks losing data...  Looking around, another possible solution would be to create a trigger that filled the column instead of leaning on the generated column, but I'd prefer to use the generated column if we can stomach the table locks.

@gigxz ?

Reply:
To reduce potential downtime, this would probably have to be two deploys: a initial migration just adds the new slugs (trigger probably). Once the slugs are built we'd update the view and change the rails associations with another deploy